File: src/main/java/com/gmail/borlandlp/minigamesdtools/arena/team/ExampleTeamListener.java
package com.gmail.borlandlp.minigamesdtools.arena.team;

import com.gmail.borlandlp.minigamesdtools.Debug;
import com.gmail.borlandlp.minigamesdtools.MinigamesDTools;
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener;
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*;
import com.gmail.borlandlp.minigamesdtools.arena.team.lobby.ArenaLobby;
import org.bukkit.entity.Player;

public class ExampleTeamListener implements ArenaEventListener {
    private TeamController teamController;

    public ExampleTeamListener(TeamController teamController) {
        this.teamController = teamController;
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.LOWEST
    )
    public void onPlayerDeath(ArenaPlayerDeathLocalEvent event) {
        this.handlePlayerDeath(event.getPlayer());
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.LOWEST
    )
    public void onPlayerKilled(ArenaPlayerKilledLocalEvent event) {
        this.handlePlayerDeath(event.getPlayer());
        Debug.print(Debug.LEVEL.NOTICE, this.getClass().getSimpleName() + "#" + event);
    }

    private void handlePlayerDeath(Player p) {
        TeamProvider teamProvider = this.teamController.getTeamOf(p);
        ArenaPlayerRespawnRequestLocalEvent arenaPlayerRespawnRequestLocalEvent = new ArenaPlayerRespawnRequestLocalEvent(teamProvider, p);
        teamProvider.getArena().getEventAnnouncer().announce(arenaPlayerRespawnRequestLocalEvent);
        if(!arenaPlayerRespawnRequestLocalEvent.isCancelled()) {
            if(((ArenaLobby)teamProvider.getRespawnLobby()).isEnabled()) {
                teamProvider.movePlayerTo((ArenaLobby) teamProvider.getRespawnLobby(), p);
            } else {
                teamProvider.spawn(p);
            }
        } else {
            teamProvider.setSpectate(p, true);
        }
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.LOWEST
    )
    public void onPlayerLeave(ArenaPlayerLeaveLocalEvent event) {
        Player player = event.getPlayer();
        TeamProvider team = this.teamController.getTeamOf(player);
        if(team != null) {
            player.getInventory().setArmorContents(null);
            player.getInventory().clear();
            team.removePlayer(player);
        }
        MinigamesDTools.Companion.getInstance().getHotbarAPI().unbindHotbar(event.getPlayer());
    }

    @ArenaEventHandler(
            ignoreCancelled = true,
            priority = ArenaEventPriority.LOWEST
    )
    public void onPlayerJoin(ArenaPlayerJoinLocalEvent event) {
        event.getTeam().addPlayer(event.getPlayer());
    }
}

File: src/main/kotlin/com/gmail/borlandlp/minigamesdtools/arena/team/ExampleTeamListener.kt
package com.gmail.borlandlp.minigamesdtools.arena.team

import com.gmail.borlandlp.minigamesdtools.Debug
import com.gmail.borlandlp.minigamesdtools.Debug.print
import com.gmail.borlandlp.minigamesdtools.MinigamesDTools.Companion.instance
import com.gmail.borlandlp.minigamesdtools.arena.ArenaEventListener
import com.gmail.borlandlp.minigamesdtools.arena.localevent.*
import com.gmail.borlandlp.minigamesdtools.arena.team.lobby.ArenaLobby
import org.bukkit.entity.Player

class ExampleTeamListener(private val teamController: TeamController) : ArenaEventListener {
    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.LOWEST)
    fun onPlayerDeath(event: ArenaPlayerDeathLocalEvent) {
        handlePlayerDeath(event.player)
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.LOWEST)
    fun onPlayerKilled(event: ArenaPlayerKilledLocalEvent) {
        handlePlayerDeath(event.player)
        print(
            Debug.LEVEL.NOTICE,
            this.javaClass.simpleName + "#" + event
        )
    }

    private fun handlePlayerDeath(p: Player) {
        val teamProvider = teamController.getTeamOf(p)
        val arenaPlayerRespawnRequestLocalEvent =
            ArenaPlayerRespawnRequestLocalEvent(teamProvider!!, p)
        teamProvider.arena.eventAnnouncer.announce(arenaPlayerRespawnRequestLocalEvent)
        if (!arenaPlayerRespawnRequestLocalEvent.isCancelled) {
            if ((teamProvider.respawnLobby as ArenaLobby).isEnabled) {
                teamProvider.movePlayerTo((teamProvider.respawnLobby as ArenaLobby), p)
            } else {
                teamProvider.spawn(p)
            }
        } else {
            teamProvider.setSpectate(p, true)
        }
    }

    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.LOWEST)
    fun onPlayerLeave(event: ArenaPlayerLeaveLocalEvent) {
        val player = event.player
        val team = teamController.getTeamOf(player)
        if (team != null) {
            player.inventory.armorContents = null
            player.inventory.clear()
            team.removePlayer(player)
        }
        instance!!.hotbarAPI!!.unbindHotbar(event.player)
    }

    @ArenaEventHandler(ignoreCancelled = true, priority = ArenaEventPriority.LOWEST)
    fun onPlayerJoin(event: ArenaPlayerJoinLocalEvent) {
        event.team.addPlayer(event.player)
    }

}